tests/Dfc.CourseDirectory.WebV2.Tests/FeatureTests/Venues/ViewVenues/ViewVenuesTests.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Dfc.CourseDirectory.Core.Models;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Microsoft.AspNetCore.Http;
+using Xunit;
+
+namespace Dfc.CourseDirectory.WebV2.Tests.FeatureTests.Venues.ViewVenues
+{
+    public class ViewVenuesTests : MvcTestBase
+    {
+        public ViewVenuesTests(CourseDirectoryApplicationFactory factory)
+            : base(factory)
+        {
+        }
+
+        [Fact]
+        public async Task ViewVenues_Get_WithNoVenues_ReturnsExpectedContent()
+        {
+            //Arange
+            var providerId = await TestData.CreateProvider(
+                providerType: ProviderType.FE | ProviderType.Apprenticeships | ProviderType.TLevels);
+
+            var request = new HttpRequestMessage(HttpMethod.Get, $"/venues?providerId={providerId}");
+
+            // Act
+            var response = await HttpClient.SendAsync(request);
+
+            // Assert
+            response.StatusCode.Should().Be(StatusCodes.Status200OK);
+
+            var doc = await response.GetDocument();
+            doc.Body.TextContent.Should().Contain("You have no locations");
```

Comment:
I'm not sure what your point is? We use data-testid everywhere in V2 and JG wrote a tag helper to strip these out in release builds: https://github.com/SkillsFundingAgency/dfc-coursedirectory/blob/main/src/Dfc.CourseDirectory.WebV2/TagHelpers/RemoveTestDataAttributesTagHelper.cs

Reply:
My point is that this isn't using `GetElementByTestId` to find the element containing the expected text and I think it should be.

Reply:
I understand you now. Your suggestion actually makes the test more fragile, as the assertion isn't just that the page tells you that information, but that it does so in a specific way. Do we care [for the purpose of this test] how that information is rendered, as long as it is rendered?

Reply:
I agree it makes the test more fragile, however the point of the ids is so that when you are working in pure markup you can see at a glance that you'll be affecting tests as you modify the markup. Without the `data-testid` you'll be blind to that till you break the test. For me that trade-off is well worth it.

Here's the original article on `data-js` which is the basis for `data-testid` and explains the value in not just hooking non-css things to the existing dom and classes: https://ultimatecourses.com/blog/data-js-selectors-enhancing-html5-development-by-separating-css-from-javascript